catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/QueryFilterTransformerHelper.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.xml.namespace.QName;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+
+import ddf.catalog.transform.QueryFilterTransformer;
+
+public class QueryFilterTransformerHelper {
+    private Map<QName, QueryFilterTransformer> queryFilterTransformerMap =
+            new ConcurrentHashMap<>();
+
+    public void bind(ServiceReference<QueryFilterTransformer> reference) {
+        if (reference == null) {
+            return;
+        }
+
+        List<QName> namespaces = getNamespaces(reference);
+        QueryFilterTransformer transformer = getTransformer(reference);
+
+        for (QName namespace : namespaces) {
+            queryFilterTransformerMap.put(namespace, transformer);
+        }
+    }
+
+    public void unbind(ServiceReference<QueryFilterTransformer> reference) {
+        if (reference == null) {
+            return;
+        }
+
+        List<QName> namespaces = getNamespaces(reference);
+
+        for (QName namespace : namespaces) {
+            queryFilterTransformerMap.remove(namespace);
+        }
+
+        getBundleContext().ungetService(reference);
```

Comment:
Done.

The `ungetService` is just for reference counting of the `ServiceReference` once the count gets to 0, the bundle context will destroy the service. I don't think its mandatory, as we use blueprint to manage most of our references and just work with the actual service objects.